system/Security/Security.php

```diff
@@ -329,7 +340,33 @@ private function getPostedToken(RequestInterface $request): ?string
      */
     public function getHash(): ?string
     {
-        return $this->hash;
+        return $this->tokenRandomize ? $this->randomize($this->hash) : $this->hash;
+    }
+
+    /**
+     * Randomize hash to avoid BREACH attacks.
+     */
+    protected function randomize(string $hash): string
+    {
+        $key   = random_bytes(self::CSRF_HASH_BYTES);
```

Comment:
`CSRF_HASH_BYTES` is set to public. If this is intended not to be overridden, then use private instead so that we can retain `self::`. Otherwise, if override is intended, then use `static::`

Reply:
It should be able to be overridden, assuming the use case that users want to increase the value.

Fixed.